Source/WebCore/html/HTMLMediaElement.cpp

```diff
@@ -6768,7 +6768,11 @@ void HTMLMediaElement::enterFullscreen(VideoFullscreenMode mode)
     m_changingVideoFullscreenMode = true;
 
 #if ENABLE(FULLSCREEN_API) && ENABLE(VIDEO_USES_ELEMENT_FULLSCREEN)
-    if (document().settings().fullScreenEnabled() && mode == VideoFullscreenModeStandard) {
+    bool videoUsesElementFullscreen = true;
+#if PLATFORM(IOS_FAMILY)
+    videoUsesElementFullscreen &= document().settings().videoFullscreenRequiresElementFullscreen();
```

Comment:
is the point of using `&=` to avoid a compiler warning that the value set to above is unused? Would it be more clear to say this?
```
#if PLATFORM(IOS_FAMILY)
bool videoUsesElementFullscreen = ...;
#else
constexpr bool videoUsesElementFullscreen = true;
#endif
```
but does not matter to me so much